library/Rules/Date.php

```diff
@@ -49,8 +49,39 @@ public function validate($input)
             $this->format = $exceptionalFormats[$this->format];
         }
 
-        $info = date_parse_from_format($this->format, $inputString);
+        return $this->isValidDateTime($input);
```

Comment:
I'd do a few more changes in this code.

- The code from 43 to 50 I'd extract to a private `fixFormatForExceptionalCases()` method.
- I'd rename 'isValidDateTime' to 'isValidForFormatProvided' and I'd use `fixFormatForExceptionalCases` inside of it.

Reply:
I don't wanna change much because on `master` this rule is completely different

Reply:
I renamed `isValidDateTime` to `isValidForFormatProvided`, the `fixFormatForExceptionalCases` shouldn't even be there because we shouldn't change the state of the rule after the validation but we're doing it here, see: https://github.com/Respect/Validation/blob/master/library/Helpers/DateTimeHelper.php#L31-L41